src/CalcViewModel/GraphingCalculator/EquationViewModel.cpp

```diff
@@ -1,14 +1,316 @@
+            SetKeyGraphFeaturesItems(m_resourceLoader->GetString(L"Minima"), Minima, m_resourceLoader->GetString(L"KGFMinimaNone"));
+            SetKeyGraphFeaturesItems(m_resourceLoader->GetString(L"Maxima"), Maxima, m_resourceLoader->GetString(L"KGFMaximaNone"));
+            SetKeyGraphFeaturesItems(
+                m_resourceLoader->GetString(L"InflectionPoints"), InflectionPoints, m_resourceLoader->GetString(L"KGFInflectionPointsNone"));
+            SetKeyGraphFeaturesItems(
+                m_resourceLoader->GetString(L"VerticalAsymptotes"), VerticalAsymptotes, m_resourceLoader->GetString(L"KGFVerticalAsymptotesNone"));
+            SetKeyGraphFeaturesItems(
+                m_resourceLoader->GetString(L"HorizontalAsymptotes"), HorizontalAsymptotes, m_resourceLoader->GetString(L"KGFHorizontalAsymptotesNone"));
+            SetKeyGraphFeaturesItems(
+                m_resourceLoader->GetString(L"ObliqueAsymptotes"), ObliqueAsymptotes, m_resourceLoader->GetString(L"KGFObliqueAsymptotesNone"));
+            SetParityStringProperty();
+            SetPeriodicityStringProperty();
+            SetMonotoncityStringProperty();
+            SetTooComplexFeaturesErrorProperty();
+
+            AnalysisErrorVisible = false;
+            IsAnalysisUpdated = false;
+        }
+    }
+
+    void EquationViewModel::SetKeyGraphFeaturesItems(String ^ title, String ^ expression, String ^ errorString)
+    {
+        KeyGraphFeaturesItem ^ item = ref new KeyGraphFeaturesItem();
+        item->Title = title;
+        if (expression != L"")
+        {
+
+            item->DisplayItems->Append(expression);
+            item->IsText = false;
+        }
+        else
+        {
+            item->DisplayItems->Append(errorString);
+            item->IsText = true;
+        }
+        KeyGraphFeaturesItems->Append(item);
+    }
+
+    void EquationViewModel::SetKeyGraphFeaturesItems(String ^ title, IObservableVector<String ^> ^ expressionVector, String ^ errorString)
+    {
+        KeyGraphFeaturesItem ^ item = ref new KeyGraphFeaturesItem();
+        item->Title = title;
+        if (expressionVector->Size != 0)
+        {
+            for (auto expression : expressionVector)
+            {
+                item->DisplayItems->Append(expression);
+            }
+            item->IsText = false;
+        }
+        else
+        {
+            item->DisplayItems->Append(errorString);
+            item->IsText = true;
+        }
+        KeyGraphFeaturesItems->Append(item);
+    }
+
+    void EquationViewModel::SetParityStringProperty()
     {
+        KeyGraphFeaturesItem ^ parityItem = ref new KeyGraphFeaturesItem();
+        parityItem->Title = m_resourceLoader->GetString(L"Parity");
+        switch (Parity)
+        {
+        case 0:
+            parityItem->DisplayItems->Append(m_resourceLoader->GetString(L"KGFParityUnknown"));
+            parityItem->IsText = true;
```

Comment:
You can factor out the partyItem->IsText = true; to outside the switch statement.